code/game/gamemodes/game_mode.dm

```diff
@@ -283,6 +283,61 @@
 		set_security_level(SEC_LEVEL_BLUE)
 
 
+// This is a frequency selection system. You may imagine it like a raffle where each player can have some number of tickets. The more tickets you have the more likely you are to
+// "win". The default is 100 tickets. If no players use any extra tickets (earned with the antagonist rep system) calling this function should be equivalent to calling the normal
+// pick() function. By default you may use up to 100 extra tickets per roll, meaning at maximum a player may double their chances compared to a player who has no extra tickets.
+//
+// The odds of being picked are simply (your_tickets / total_tickets). Suppose you have one player using fifty (50) extra tickets, and one who uses no extra:
+//     Player A: 150 tickets
+//     Player B: 100 tickets
+//        Total: 250 tickets
+//
+// The odds become:
+//     Player A: 150 / 250 = 0.6 = 60%
+//     Player B: 100 / 250 = 0.4 = 40%
+/datum/game_mode/proc/antag_pick(list/datum/candidates)
+	if(!CONFIG_GET(flag/use_antag_rep)) // || candidates.len <= 1)
+		return pick(candidates)
+
+	// Tickets start at 100
+	var/DEFAULT_ANTAG_TICKETS = CONFIG_GET(number/default_antag_tickets)
+
+	// You may use up to 100 extra tickets (double your odds)
+	var/MAX_TICKETS_PER_ROLL = CONFIG_GET(number/max_tickets_per_roll)
+
+
+	var/total_tickets = 0
+
+	MAX_TICKETS_PER_ROLL += DEFAULT_ANTAG_TICKETS
+
+	var/p_ckey
+	var/p_rep
+
+	for(var/datum/mind/mind in candidates)
+		p_ckey = ckey(mind.key)
+		total_tickets += min(SSpersistence.antag_rep[p_ckey] + DEFAULT_ANTAG_TICKETS, MAX_TICKETS_PER_ROLL)
+
+	var/antag_select = rand(1,total_tickets)
+	var/current = 1
+
+	for(var/datum/mind/mind in candidates)
+		p_ckey = ckey(mind.key)
+		p_rep = SSpersistence.antag_rep[p_ckey]
+		p_rep = p_rep == null ? 0 : p_rep
+
+		if(current <= antag_select)

```

Comment:
current = 1
antag_select = rand(1 to total_tickets)
current is always <= antag_select

if somehow this if statement isn't hit, current goes through a += with I assume >0 values so it would never hit this if statement again.

In other words this either always returns the first mind or never returns a mind.

Reply:
@optimumtact I don't know if you're still subscribed to this PR so I'm going to annoy the hell out of you because I care.

Reply:
Let's step through it manually. Suppose you have five players and each of them has only one ticket:

1 playerA
1 playerB
1 playerC
1 playerD
1 playerE

That means:

total_tickets = 5
current = 1
antag_select = 3 (a value from `rand(1 to 5)`)

Looping through the above mind list:

```
Iterating: playerA
	current: 1

	current (1) <= antag_select (3) FALSE

	current += playerA's tickets (1)
-----------------------------------------------------
Iterating: playerB
	current: 2

	current (2) <= antag_select (3) FALSE

	current += playerA's tickets (1)
-----------------------------------------------------
Iterating: playerC
	current: 3

	current (3) <= antag_select (3) TRUE	

	--> return playerC
```

This should work properly, but in the event it doesn't you'll note right under that it should show a warning in the server console and skip the antag system entirely:

```dm
	WARNING("Something has gone terribly wrong. /datum/game_mode/proc/antag_pick failed to select a candidate. Falling back to pick()")
	return pick(candidates)
```

Reply:
I'm not losing my mind am I?
if(1 <= 3)
if(1 is less than or equal to 3)
if(1 is not greater than 3)

Reply:
@CosmicScientist ... no you're not actually. It's supposed to be reversed. I'm not braining right now. Thanks.

Reply:
@CosmicScientist Should be fixed in #36188. Please read over that and tell me if _I'm_ losing my mind before it's merged.

Reply:
It's okay. You're making it work right now and that's what matters!
❤️